src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/ItemGroupIntrinsicTask.cs

```diff
@@ -534,17 +547,24 @@ private List<ProjectItemInstance> FindItemsMatchingSpecification
                 // wildcards.  Then loop through each file returned, and add it
                 // to our hashtable.
 
-                // Don't unescape wildcards just yet - if there were any escaped, the caller wants to treat them
-                // as literals. Everything else is safe to unescape at this point, since we're only matching
-                // against the file system.
-                string[] fileList = EngineFileUtilities.GetFileListEscaped(Project.Directory, piece);
+                try
+                {
+                    // Don't unescape wildcards just yet - if there were any escaped, the caller wants to treat them
+                    // as literals. Everything else is safe to unescape at this point, since we're only matching
+                    // against the file system.
+                    string[] fileList = EngineFileUtilities.GetFileListEscaped(Project.Directory, piece);
 
-                foreach (string file in fileList)
+                    foreach (string file in fileList)
+                    {
+                        // Now unescape everything, because this is the end of the road for this filename.
+                        // We're just going to compare it to the unescaped include path to filter out the
+                        // file excludes.
+                        specificationsToFind.Add(EscapingUtilities.UnescapeAll(file));
+                    }
+                }
+                catch (DriveEnumerationWildcardException ex)
```

Comment:
Test:
```
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
    <Target Name="TestTarget">
        <ItemGroup>
            <FilesToCopy Include="$(Microsoft_WindowsAzure_EngSys)\*.pdb" Exclude="$(Microsoft_WindowsAzure_EngSys)\**\*" />
        </ItemGroup>
    </Target>
</Project>
```
Output:
![image](https://user-images.githubusercontent.com/89041446/144317746-80d2be93-6fcd-49af-97bb-d67b6a37e080.png)
Is this test case invalid, or is this an internal error that I should investigate further to debug this scenario?

Reply:
New test (following output appears for both new and old test):
```
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
    <Target Name="TestTarget">
        <ItemGroup>
            <FilesToCopy Include="c:\src\*.log" Exclude="$(Microsoft_WindowsAzure_EngSys)\**\*" />
        </ItemGroup>
    </Target>
</Project>
```
Output with new changes:
![image](https://user-images.githubusercontent.com/89041446/144525095-699ddd59-0f39-4ef0-a894-3bdc92e05056.png)